src/components/admin/category/CategoryList.js

```diff
@@ -7,11 +9,12 @@ class CategoryList extends Component {
         categories: []
     }
 
-    componentDidMount() {
-        this.setState((prevState) => ({ categories: [{ id: 1, name: "Electronics" }, { id: 2, name: "Books" }, { id: 3, name: "Clothes" }] }))
+    async componentDidMount() {
+        const categories = await getCategories() //resolved promise
+        this.setState((prevState) => ({ categories: categories.data }))
```

Comment:
again: I don't see `async componentDidMount` as bad, I just haven't seen it done before
https://stackoverflow.com/questions/47970276/is-using-async-componentdidmount-good

can getCategories just return .data?